tests/csapi/account_data_test.go

```diff
@@ -0,0 +1,83 @@
+package csapi_tests
+
+import (
+	"testing"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+)
+
+func TestAddAccountData(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintOneToOneRoom)
+	defer deployment.Destroy(t)
+
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+
+	// sytest: Can add account data
+	// sytest: Can get account data without syncing
+	t.Run("Can add global account data", func(t *testing.T) {
+		// Set the account data entry
+		resp := alice.SetGlobalAccountData(t, "com.beeper.test", map[string]interface{}{"value": "first"})
+		must.MatchResponse(t, resp, match.HTTPResponse{
+			StatusCode: 200,
+		})
```

Comment:
[`SetGlobalAccountData`](https://github.com/matrix-org/complement/blob/cdad7e9d553ae6169c88a35c7a3d6d37c160397d/internal/client/client.go#L179-L181) makes use of a [`MustDoFunc`](https://github.com/matrix-org/complement/blob/cdad7e9d553ae6169c88a35c7a3d6d37c160397d/internal/client/client.go#L474-L484) call, which will already fail the test if the status code is not 2xx.

Likewise with the `GetGlobalAccountData` calls, and the get/set room variants below.